test/test_type_promotion.py

```diff
@@ -312,6 +312,64 @@ def test_lt_with_type_promotion(self):
             self.assertTrue(actual, expected)
             self.assertTrue(actual.dtype == torch.bool)
 
+    def test_sparse(self):
```

Comment:
If I'm reading the code correctly, for add and mul, type promotion is implemented for the following:
- add_out_sparse (cpu + cuda)
- add_out_sparse_dense_sparse (cpu only)
- mul_out_sparse (cpu + cuda)

Why is there a discrepancy in what is supported in `add_out_sparse_dense_sparse`?